ui/shared/components/form/FormWizard.jsx

```diff
@@ -26,31 +29,37 @@ class FormWizard extends React.PureComponent {
     }))
   }
 
-  onPageSubmitSucceeded = values => this.setState(prevState => ({
-    asyncValues: { ...prevState.asyncValues, ...(values || {}) },
-  }))
-
-  resolvedPageSubmit = () => Promise.resolve()
+  onPageSubmit = (url, formatUrl) => values => (
+    new HttpRequestHelper(
+      url || formatUrl(values),
+      newValues => this.setState(prevState => ({
+        asyncValues: { ...prevState.asyncValues, ...(newValues || {}) },
+      })),
+    ).post(values)
+  )
 
   setSubmitSucceeded = () => this.setState({ formSubmitSucceeded: true })
 
   onFormSubmit = (values) => {
-    const { onSubmit } = this.props
+    const { formatSubmitUrl, formatSubmitValues, onSubmitSuccess } = this.props
     const { asyncValues } = this.state
-    return onSubmit({ ...asyncValues, ...values })
+    const allValues = { ...asyncValues, ...values }
+    return new HttpRequestHelper(
```

Comment:
since all usages for submitting the form are to trigger a request, it seemed neater to move the request handling here